### PR TITLE
Identity refactor/naming

### PR DESCRIPTION
**Pull request by ericjordanmossman**
This pull request refactors the name of several classes and methods related to MongoDB Realm. It's intended to cover [RCOCOA-677](https://jira.mongodb.org/browse/RCOCOA-677) and [RCOCOA-762](https://jira.mongodb.org/browse/RCOCOA-762)

- A Realm user's "identity" string is renamed to Identifier in order differentiate from a user's RLMUserIdentity. 
- Adds `User.isLoggedIn`
- Other minor renames to make Cocoa sdk more consistent with other Realm SDKs
- Updates the assisted_aggs and /baas version in the Object Server Tests.

### CODE REVIEW

**Comment by ericjordanmossman** [comment]
I'm moving out of draft with the following failures. When reviewing could you please also check to make sure I'm interpreting these failures correctly.

**Jenkins: ios-swift Release 12.0**
The failing tests are
<details>


```
Sep 09 09:17:33 CombinePublisherTests<Results>
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("1") is not equal to ("0")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:654
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.isFrozen, frozen)
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("[0]") is not equal to ("[-1]")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:655
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33             XCTAssertEqual(deletions, [])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("2") is not equal to ("1")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:654
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.isFrozen, frozen)
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("[1]") is not equal to ("[0]")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:655
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33             XCTAssertEqual(deletions, [])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("3") is not equal to ("2")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:654
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.isFrozen, frozen)
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("[2]") is not equal to ("[1]")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:655
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33             XCTAssertEqual(deletions, [])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("4") is not equal to ("3")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:654
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.isFrozen, frozen)
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("[3]") is not equal to ("[2]")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:655
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33             XCTAssertEqual(deletions, [])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("5") is not equal to ("4")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:654
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.isFrozen, frozen)
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("[4]") is not equal to ("[3]")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:655
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33             XCTAssertEqual(deletions, [])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("6") is not equal to ("5")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:654
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.isFrozen, frozen)
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("[5]") is not equal to ("[4]")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:655
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33             XCTAssertEqual(deletions, [])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("7") is not equal to ("6")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:654
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.isFrozen, frozen)
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("[6]") is not equal to ("[5]")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:655
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33             XCTAssertEqual(deletions, [])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("8") is not equal to ("7")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:654
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.isFrozen, frozen)
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("[7]") is not equal to ("[6]")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:655
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33             XCTAssertEqual(deletions, [])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("9") is not equal to ("8")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:654
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.isFrozen, frozen)
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33   testFrozenChangeSetSubscribeOn, XCTAssertEqual failed: ("[8]") is not equal to ("[7]")
Sep 09 09:17:33   /Users/realm/workspace/objc_pr/configuration/Release/target/ios-swift/xcode_version/12.0/RealmSwift/Tests/CombineTests.swift:655
Sep 09 09:17:33   ```
Sep 09 09:17:33             XCTAssertEqual(collection.count, calls)
Sep 09 09:17:33             XCTAssertEqual(insertions, [calls - 1])
Sep 09 09:17:33             XCTAssertEqual(deletions, [])
Sep 09 09:17:33   ```
Sep 09 09:17:33 
Sep 09 09:17:33 
Sep 09 09:17:33 	 Executed 1639 tests, with 18 failures (0 unexpected) in 12.878 (13.891) seconds
```

</details>


I think this could be a cleaning issue from the previous CI build ([which succeeded](https://ci.realm.io/job/objc_pr/7017/configuration=Release,target=ios-swift,xcode_version=12.0/console)): 
Note how the equality is always one more than expected. Also the collection iOS-swift tests pass locally. 


**Jenkins: osx-object-server Release 11.3**
11 failing tests

_Sep 09 09:01:22     ✗ testDownloadRealmToCustomPath, Test crashed_ (fails in base, see https://github.com/realm/realm-cocoa/issues/6719)
_Sep 09 08:52:13     ✗ testClientReset, (([pathValue rangeOfString:recoveryPath].location != NSNotFound) is true) failed
Sep 09 09:01:03     ✗ testClientReset, XCTAssertTrue failed_ (I believe these are holdovers from how client reset works in legacy Cloud. They fail in base.
)

The following fail in wait, not the primary assertions. They all pass locally. I attribute them to potential flakiness.

_Sep 09 08:54:03     ✗ testMultipleWatchStreams, Asynchronous wait failed: Exceeded timeout of 60 seconds, with unfulfilled expectations: "watch collection and receive change event 3 times"._ (Passes in [#7017](https://ci.realm.io/job/objc_pr/7017/configuration=Release,target=osx-object-server,xcode_version=11.3/consoleFull))
_Sep 09 08:56:20     ✗ testWatch, Asynchronous wait failed: Exceeded timeout of 60 seconds, with unfulfilled expectations: "watch collection and receive change event 3 times"._
_Sep 09 08:57:21     ✗ testWatchAsync, Asynchronous wait failed: Exceeded timeout of 60 seconds, with unfulfilled expectations: "watch collection and receive change event 3 times"._ (Passes in #7017)
_Sep 09 08:59:58     ✗ testWatchWithMatchFilter, Asynchronous wait failed: Exceeded timeout of 60 seconds, with unfulfilled expectations: "watch collection and receive change event 3 times"._ (Passes in #7017)
_Sep 09 09:03:52     ✗ testWatchAsync, Asynchronous wait failed: Exceeded timeout of 60 seconds, with unfulfilled expectations: "Watch 3 document events".
Sep 09 09:08:46     ✗ testWatchWithFilterIds, Asynchronous wait failed: Exceeded timeout of 60 seconds, with unfulfilled expectations: "Watch 3 document events"._ (Passes in #7017)
_Sep 09 09:09:47     ✗ testWatchWithFilterIdsAsync, Asynchronous wait failed: Exceeded timeout of 60 seconds, with unfulfilled expectations: "Watch 3 document events"._ (Passes in #7017)
_Sep 09 09:10:49     ✗ testWatchWithMatchFilter, Asynchronous wait failed: Exceeded timeout of 60 seconds, with unfulfilled expectations: "Watch 3_ (Passes in #7017)

**Jenkins: osx-object-server Release 12.0**
Similar to the 11.3 object-server tests above.

_Sep 09 09:12:21 Failing tests:
Sep 09 09:12:21 	ObjectServerTests:
Sep 09 09:12:21 		-[RLMObjectServerTests testClientReset]_ **(Fails in base, legacy client reset)**
_Sep 09 09:12:21 		-[RLMObjectServerTests testMultipleWatchStreams]
Sep 09 09:12:21 		-[RLMObjectServerTests testWatch]
Sep 09 09:12:21 		SwiftObjectServerTests.testClientReset()_  **(Fails in base, legacy client reset)**
_Sep 09 09:12:21 		SwiftObjectServerTests.testDownloadRealmToCustomPath()_ **(Fails in base, known crash https://github.com/realm/realm-cocoa/issues/6719)**
_Sep 09 09:12:21 		SwiftObjectServerTests.testWatch()
Sep 09 09:12:21 		SwiftObjectServerTests.testWatchAsync()_

Watch tests pass locally and in other builds. They only fail in CI due to timeouts, not primary assertion failure.
Screen Shot 2020-09-09 at 3 16 45 PM

**Jenkins: swiftpm-ios Release 12.0**
This aborts while building a util downloaded from brew.
Passes in #7017 (https://ci.realm.io/job/objc_pr/7017/configuration=Release,target=swiftpm-ios,xcode_version=12.0/), likely a CI clean issue?

**Jenkins: tvos-swift Release 12.0**
This fails because CI machine ran out of space
Sep 09 09:07:08 ❌  fatal error: error in backend: IO failure on output stream: No space left on device
Also passed in first build #7017: https://ci.realm.io/job/objc_pr/7017/configuration=Release,target=tvos,xcode_version=12.0/

**Jenkins: watchos Release 12.0**
Same issues as swiftpm-ios Release 12.0. It looks like crashes after brew setup:
```
Warning: moreutils 0.64 is already installed and up-to-date
To reinstall 0.64, run `brew reinstall moreutils`
+ ./build.sh ci-pr
+ ts
Sep 09 08:52:11 Killing running Simulator processes... done!
Sep 09 08:52:11 Shut down existing simulator devices... done!
Sep 09 08:53:48 Deleting all simulators... done!
Sep 09 08:55:48 Creating fresh simulators... done!
Sep 09 08:56:02 Booting iPhone 8 simulator... done!
Build timed out (after 45 minutes). Marking the build as aborted.
Build was aborted
```

Also succeeds in #7017 (https://ci.realm.io/job/objc_pr/7017/configuration=Release,target=watchos,xcode_version=12.0/)

File: Realm/ObjectServerTests/setup_baas.rb
Comment (leemaguire) [inline Realm/ObjectServerTests/setup_baas.rb:105]:
```suggestion
        puts `cd '#{stitch_dir}' && curl --silent -O "https://s3.amazonaws.com/stitch-artifacts/stitch-mongo-libs/stitch_mongo_libs_osx_patch_b1c679a26ecb975372de41238ea44e4719b8fbf0_5f3d91c10ae6066889184912_20_08_19_20_57_17/assisted_agg"`
```

File: Realm/RLMEmailPasswordAuth.h
Comment (leemaguire) [inline Realm/RLMEmailPasswordAuth.h:44]: This should be `registerUserWithEmail`, also indentation needs correction

File: Realm/RLMUser.mm
Comment (leemaguire) [inline Realm/RLMUser.mm:153]: indentation

**Review by leemaguire — CHANGES_REQUESTED** [review]
Looks good, I've added some minor comments.

File: Realm/RLMEmailPasswordAuth.h
Comment (tgoyne) [inline Realm/RLMEmailPasswordAuth.h:44]: This function needs `NS_SWIFT_NAME(registerUser(email:password:completion:)` so that it gets bridged into swift sensibly (rather than the default of having the first label be `withEmail:`).

File: Realm/RLMEmailPasswordAuth.mm
Comment (tgoyne) [inline Realm/RLMEmailPasswordAuth.mm:32]: Indentation of following lines need to be updated.

**Comment by tgoyne** [comment]
Needs changelog entries as well.

File: CHANGELOG.md
Comment (jsflax) [inline CHANGELOG.md:4]:
```suggestion
* `Credentials(username:password:)` is renamed to `Credentials(email:password:)`
```

File: CHANGELOG.md
Comment (jsflax) [inline CHANGELOG.md:8]:
```suggestion
* Refactor `Credentials` and `User` methods for MongoDB Realm consistency.
```

File: Realm/ObjectServerTests/RLMSyncTestCase.mm
Comment (jsflax) [inline Realm/ObjectServerTests/RLMSyncTestCase.mm:190]: This line should be correctly indented.

**Review by jsflax — APPROVED** [review]

**Review by leemaguire — APPROVED** [review]

**Comment by ericjordanmossman** [comment]
@leemaguire @jsflax 
Sorry I needed to ask for another review because I accidentally reintroduced RLMUserAccountInfo and RLMUserInfo after merging. So it's useful for another quick look over in case I missed something else..

**Review by leemaguire — APPROVED** [review]

File: CHANGELOG.md
Comment (tgoyne) [inline CHANGELOG.md:4]: This isn't the only rename done in this PR and they all should be listed.

File: CHANGELOG.md
Comment (tgoyne) [inline CHANGELOG.md:8]: This isn't an accurate or useful changelog message. There isn't any refactoring done in this PR (it's all just renaming things), and if there was that wouldn't normally get a changelog entry because it isn't relevant to users of the SDK.

File: CHANGELOG.md
Comment (tgoyne) [inline CHANGELOG.md:24]:
```suggestion
| `-[RLMUser apiKeyAuth]`                                                 | `-[RLMUser apiKeysAuth]`                                                          |
```

File: CHANGELOG.md
Comment (tgoyne) [inline CHANGELOG.md:26]:
```suggestion
| `-[RLMEmailPasswordAuth registerEmail:password:completion:]`                         | `-[RLMEmailPasswordAuth registerUserWithEmail:password:completion:]`                                         |
```

**Review by tgoyne — APPROVED** [review]

**Pull request merged**
